xarray/core/utils.py

```diff
@@ -660,6 +662,30 @@ def __len__(self) -> int:
         return len(self._data) - num_hidden
 
 
+def infix_dims(dims_supplied: Collection, dims_all: Collection) -> Generator:
```

Comment:
Since this generator will only yield values, I think `Iterator` is slightly more appropriate:
https://docs.python.org/3/library/typing.html#typing.Generator